app/javascripts/components/gather.js

```diff
@@ -1,1119 +1,1152 @@
+		}
+
+		if (gather.gatherers.length > 0) {
+			return (
+				<div>
+					<div className="panel panel-primary add-bottom">
+						<div className="panel-heading">{gather.name} ({gather.description})</div>
+						<div className="panel-body">
+							<GatherProgress gather={gather} />
+							<GatherActions gather={gather} user={user} thisGatherer={thisGatherer}
+								socket={socket} />
+						</div>
+					</div>
+					<Gatherers gather={gather} user={user} thisGatherer={thisGatherer}
+						socket={socket} soundController={soundController} />
+					{gatherTeams}
+					{voting}
+				</div>
+			);
+		} else {
+			return (
+				<div>
+					<div className="panel panel-primary add-bottom">
+						<div className="panel-heading">{gather.name} ({gather.description})</div>
+					</div>
+					<Gatherers gather={gather} user={user} thisGatherer={thisGatherer}
+						socket={socket} soundController={soundController} />
+				</div>
+			);
+		}
+
+	}
+});
+
+const LifeformIcons = exports.LifeformIcons = React.createClass({
+	availableLifeforms() {
+		return ["skulk", "gorge", "lerk", "fade", "onos", "commander"];
+	},
+
+	gathererLifeforms() {
+		let lifeforms = [];
+		let gatherer = this.props.gatherer;
+		let abilities = gatherer.user.profile.abilities;
+		for (let attr in abilities) {
+			if (abilities[attr]) lifeforms.push(_.capitalize(attr));
+		}
+		return lifeforms;
+	},
+
+	render() {
+		let lifeforms = this.gathererLifeforms();
+		let availableLifeforms = this.availableLifeforms();
+		let icons = availableLifeforms.map(lifeform => {
+			let containsAbility = lifeforms.some(gathererLifeform => {
+				return gathererLifeform.toLowerCase() === lifeform.toLowerCase()
+			});
+			if (containsAbility) {
+				return <img
+					className="lifeform-icon"
+					key={lifeform}
+					src={`/${lifeform.toLowerCase()}.png`} />
+			} else {
+				return <img
+					className="lifeform-icon"
+					key={lifeform}
+					src={`/blank.gif`} />
+			}
+		});
+		return <span className="add-right hidden-xs">{icons}</span>
+	}
+});
+
+const GatherMenu = exports.GatherMenu = React.createClass({
+	propTypes: {
+		gatherPool: React.PropTypes.object.isRequired,
+		currentGather: React.PropTypes.string.isRequired,
+		gatherSelectedCallback: React.PropTypes.func.isRequired
+	},
+
+	onClick(gather) {
+		return () => {
+			this.props.gatherSelectedCallback(gather.type);
+		}
+	},
+
+	itemClass(gather) { 
+		let className = ["treeview"];
+		if (gather.type === this.props.currentGather) {
+			className.push("active");
+		}
+		return className.join(" ");
+	},
+
+	gatherPoolArray() {
+		const gatherArray = [];
+		const gatherPool = this.props.gatherPool;
+		for (let attr in gatherPool) {
+			if (gatherPool.hasOwnProperty(attr)) {
+				gatherArray.push(gatherPool[attr]);
+			}
+		}
+		return gatherArray.sort((a, b) => a.name - b.name);
+	},
+
+	drawerItemsClass(open){
+		if(!open){
+			return 'drawerItems'
+		}
+	},
+
+	getGatherTypeIcon(name){

```

Comment:
You are tying the icons to the current set of gathers which contradicts the idea of having them easily configurable, could you weave this into the configuration maybe or have it as convention over configuration even?

Reply:
Ok I should have explored the files more.

I removed the function to get the icons in gather.js and instead added the path in gather_pool.js.

Seems to work.